packages/smart-contracts/test/contracts/ERC20TransferrableReceivable.test.ts

```diff
@@ -0,0 +1,208 @@
+      const key = ethers.utils.solidityKeccak256(['address', 'bytes'], [user1Addr, paymentRef]);
+      expect(await receivable.receivableTokenIdMapping(key)).to.equals(tokenId);
+      const info = await receivable.receivableInfoMapping(tokenId);
+      expect(info[0]).to.equals(testToken.address);
+      expect(info[1]).to.equals(BASE_DECIMAL);
+      expect(info[2]).to.equals(0);
+    });
+
+    it('list receivables', async function () {
+      await receivable.connect(user1).mint('0x01', BASE_DECIMAL, testToken.address, '1');
+      await receivable.connect(user1).mint('0x02', BASE_DECIMAL, testToken.address, '2');
+      await receivable.connect(user1).mint('0x03', BASE_DECIMAL, testToken.address, '3');
+      await verifyReceivables(user1Addr, [1, 2, 3]);
+      await receivable.connect(user2).mint('0x04', BASE_DECIMAL, testToken.address, '4');
+      await receivable.connect(user2).mint('0x05', BASE_DECIMAL, testToken.address, '5');
+      await verifyReceivables(user2Addr, [4, 5]);
+      await receivable.connect(user1).transferFrom(user1Addr, user2Addr, 1);
+      await verifyReceivables(user1Addr, [3, 2]);
+      await verifyReceivables(user2Addr, [4, 5, 1]);
+      await verifyReceivableIndexes([3, 2, 1, 1, 2]);
+      await receivable
+        .connect(user2)
+        ['safeTransferFrom(address,address,uint256)'](user2Addr, user1Addr, 5);
+      await verifyReceivables(user1Addr, [3, 2, 5]);
+      await verifyReceivables(user2Addr, [4, 1]);
+      await verifyReceivableIndexes([2, 2, 1, 1, 3]);
+      await receivable.connect(user1).approve(user3Addr, 3);
+      await receivable.connect(user1).approve(user3Addr, 2);
+      await receivable.connect(user1).approve(user3Addr, 5);
+      await receivable.connect(user2).approve(user3Addr, 4);
+      await verifyReceivableIndexes([2, 2, 1, 1, 3]);
+      await receivable.connect(user3).transferFrom(user1Addr, user3Addr, 5);
+      await receivable.connect(user3).transferFrom(user1Addr, user3Addr, 2);
+      await receivable.connect(user3).transferFrom(user1Addr, user3Addr, 3);
+      await receivable.connect(user3).transferFrom(user2Addr, user3Addr, 4);
+      await verifyReceivables(user1Addr, []);
+      await verifyReceivables(user2Addr, [1]);
+      await verifyReceivables(user3Addr, [5, 2, 3, 4]);
+      await verifyReceivableIndexes([1, 2, 3, 4, 1]);
+    });
+  });
+
+  describe('payOwner', async function () {
+    it('revert with zero amount', async function () {
+      const paymentRef = '0x01' as BytesLike;
+      await receivable.connect(user1).mint(paymentRef, BASE_DECIMAL, testToken.address, '1');
+      const ids = await receivable.getIds(await user1.getAddress());
+      const tokenId = ids[0];
+      await expect(
+        receivable.payOwner(tokenId, 0, paymentRef, 0, ethers.constants.AddressZero),
+      ).to.be.revertedWith('Zero amount provided');
+    });
+
+    it('revert with "transferFromWithReferenceAndFee failed"', async function () {
```

Comment:
It's not clear what failure we're testing here, can you rename the unit test please?

Reply:
I reopen this one, I still don't understand. What are we testing here exactly?